Chore: New workflow to add and update epic issue into parent project

meanmina opened the pull request:
Tested in https://github.com/grafana/github-actions-testrepo/blob/main/.github/workflows/issue-add-to-parent-project.yml

**What is this feature?**

Creating a new Github workflow to add & update issue items between child project onto parent project.

**Why do we need this feature?**

This allows us to keep epic status up to date on parent project Platform UX based on each individual squad project. This is currently an experiment and can be later extended to cover multiple parent projects if successful.

**Who is this feature for?**

Platform UX team members

**Special notes for your reviewer**:
If you want to see it in action, have a look at https://github.com/grafana/github-actions-testrepo/actions/workflows/issue-add-to-parent-project.yml

meanmina committed:
Create issue-add-to-parent-project.yml

Tested in https://github.com/grafana/github-actions-testrepo/blob/main/.github/workflows/issue-add-to-parent-project.yml

gelicia commented on .github/workflows/issue-add-to-parent-project.yml:
```diff
@@ -0,0 +1,134 @@
+name: When epic issues changed, check if part of specified child projects and update on specified parent project
```
If this is not a generic rule and is specific to Platform UX and the squads that make up Platform UX, it might be helpful to have that in the name here and in the yml file name. 

If another team with the same parent/child hierarchy in projects wants this action, would they have to copy this file and add their own parent/child projectIDs?

meanmina replied:
Platform UX specific now as an experiment but the plan would be to expand it if it works.
Yep they would have to copy and replace some values so maybe I should rename the doc to make it platform ux specific then for now.

meanmina replied:
Renamed and updated the description as well. Thanks!

meanmina committed:
Rename issue-add-to-parent-project

Rename issue-add-to-parent-project and update to make it clear it's Platform UX specific for now.

vickyyyyyyy commented on .github/workflows/epic-add-to-platform-ux-parent-project.yml:
```diff
@@ -0,0 +1,134 @@
+name: When epic issues changed in Platform UX squad porjects, check if epic is part of specified child projects and update on Platform UX parent project
```
small typo here: `porjects`

vickyyyyyyy commented on .github/workflows/epic-add-to-platform-ux-parent-project.yml:
```diff
@@ -0,0 +1,134 @@
+
+on:
+  issues:
+    types: [opened, edited, reopened, assigned, unassigned, labeled, unlabeled]
```
I'm not familiar with GH triggers or the specific cases you'd want to catch - would you care about when issues are closed?

meanmina replied:
That's a good shout. Initially, I was worried about not triggering the workflow too often but since we're filtering for epics only, it should be ok. I will change.

meanmina committed:
Add action trigger on closed epics

Update epic-add-to-platform-ux-parent-project.yml to add action trigger on closed alongside the others and fix typo.

vickyyyyyyy reviewed (approved):
I don't have much experience with GH actions but all looks generally good with me once the Enterprise tests are fixed 👍
